kore/test/Test/Kore/Step/Rule/Simplify.hs

```diff
@@ -223,3 +243,76 @@ runSimplifyRule rule =
     $ runSimplifier Mock.env $ do
         SMT.All.declare Mock.smtDeclarations
         simplifyRuleLhs rule
+
+test_simplifyClaimRule :: [TestTree]
+test_simplifyClaimRule =
+    [ test "infers definedness"
+        rule1
+        [rule1']
+    , test "includes side condition"
+        rule2
+        [rule2']
+    ]
+  where
+    rule1, rule2, rule2' :: RulePattern Variable
+    rule1 = rulePattern (Mock.f Mock.a) Mock.b
+    rule1' = rule1 & requireDefined
+    rule2 =
+        rulePattern @Variable (Mock.g Mock.a) Mock.b
+        & Lens.set (field @"requires") requires
+    rule2' =
+        rule2
+        & requireDefined
+        & Lens.set (field @"left") (Mock.f Mock.a)
+
+    requires :: Predicate Variable
+    requires = makeEqualsPredicate Mock.testSort Mock.a Mock.b
+
+    requireDefined rule =
+        Lens.over
+            (field @"requires")
+            (flip makeAndPredicate
+                (makeCeilPredicate sort left)
+            )
+            rule
+      where
+        left = Lens.view (field @"left") rule
+        sort = termLikeSort left
+
+    test
+        :: HasCallStack
+        => TestName
+        -> RulePattern Variable
+        -> [RulePattern Variable]
+        -> TestTree
+    test name (OnePathRule -> input) (map OnePathRule -> expect) =
```

Comment:
You can leave it as-is if you want (this is a rather small test file), but I would be tempted to make it more obvious that this test uses the `g(a) => f(a)` mapping below, possibly by sending it as an argument.